### PR TITLE
MINOR: New year code cleanup - include final keyword

### PR DESCRIPTION
**Pull request by divijvaidya**
Changes:

1. Code cleanup to include `final` keyword where ever required. Detected via automatic tooling.
2. Change the casing of variables to SNAKE_CASE which are now `static final`
3. Re-order fields so that fields with `final` qualifier are grouped together

### CODE REVIEW

File: connect/api/src/main/java/org/apache/kafka/connect/data/SchemaProjector.java [hunk 1]
```diff
@@ -37,13 +37,13 @@
 
 public class SchemaProjector {
 
-    private static Set<AbstractMap.SimpleImmutableEntry<Type, Type>> promotable = new HashSet<>();
+    private static final Set<AbstractMap.SimpleImmutableEntry<Type, Type>> PROMOTABLE = new HashSet<>();
```
Comment (vamossagar12) [inline connect/api/src/main/java/org/apache/kafka/connect/data/SchemaProjector.java:40]: nit: We should probably call out that the casing of variable(s) has been changed  in the PR description? Currently it only says the usage of `final`..

Comment (vamossagar12) [reply connect/api/src/main/java/org/apache/kafka/connect/data/SchemaProjector.java:40]: Also, wondering if this change got imposed due to a checkstyle?

**Review by vamossagar12 — APPROVED** [review]
Thanks for the PR @divijvaidya . Added 1 minor comment about the description of the PR. Also, there already exists a [checkstyle](https://github.com/apache/kafka/blob/trunk/checkstyle/checkstyle.xml#L74-L78) for the same which has been enabled only for streams. 

If we don't want to end up in this situation again, should we remove the suppressions for modules other than streams? Or would it be too big a change? Also, not sure which tool was used by you for this reporting.

File: clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java [hunk 2]
```diff
@@ -479,7 +479,7 @@ private static class NodeConnectionState {
         private final String host;
         private final HostResolver hostResolver;
         private InetAddress lastAttemptedAddress;
-        private Logger log;
+        private final Logger log;
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java]: We typically have the final fields grouped first.

Comment (divijvaidya) [reply clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java]: ack. will change.

File: connect/api/src/main/java/org/apache/kafka/connect/data/SchemaProjector.java (hunk 1 above)
Comment (divijvaidya) [reply connect/api/src/main/java/org/apache/kafka/connect/data/SchemaProjector.java:40]: Added to the description and yes, checkstyle complained about it.

**Comment by divijvaidya** [comment]
Thank you for the review @vamossagar12 . I didn't know about the checkstyle rule in streams. I will try & observe if enabling it for other modules leads to a large number of changes or not. I found this using intelliJ's code analysis.

**Comment by ijuma** [comment]
The streams final rule also includes local variables, etc (which is quite noisy). If there is a way to do it for fields only, that would be great.

File: clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java (hunk 2 above)
Comment (divijvaidya) [reply clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java]: This is done in the latest commit.

**Comment by divijvaidya** [comment]
Checkstyle doesn't have a rule [1] available to enforce that fields which are not changing are marked as final. Hence, I am not changing anything in the checkstyle here in this PR.

[1] https://checkstyle.sourceforge.io/checks.html

**Comment by vamossagar12** [comment]
Thanks @divijvaidya , I think Ismael's point is correct. From what I recall, the streams final rule also expects the method arguements etc to be final as well (apart from local variables within methods). Enabling that rule would be a big change I guess. Also, I see only variables and parameters [here](https://checkstyle.sourceforge.io/checks/coding/finallocalvariable.html#FinalLocalVariable) for tokens to check. I think we can leave it to this but the only problem is, there won't be any such enforcements for using final variables and might need cleanups in the future. That should be ok IMO.

File: clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java [hunk 3]
```diff
@@ -64,11 +64,11 @@ public class MockConsumer<K, V> implements Consumer<K, V> {
     private final Map<TopicPartition, OffsetAndMetadata> committed;
     private final Queue<Runnable> pollTasks;
     private final Set<TopicPartition> paused;
+    private final AtomicBoolean wakeup;
 
-    private Map<TopicPartition, List<ConsumerRecord<K, V>>> records;
+    private final Map<TopicPartition, List<ConsumerRecord<K, V>>> records;
```
Comment (mimaison) [inline clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java:69]: Let's add the empty line after this field to separate the final fields from the other ones.

File: clients/src/main/java/org/apache/kafka/common/record/LazyDownConversionRecordsSend.java [hunk 4]
```diff
@@ -36,9 +36,9 @@ public final class LazyDownConversionRecordsSend extends RecordsSend<LazyDownCon
     private static final int MAX_READ_SIZE = 128 * 1024;
     static final int MIN_OVERFLOW_MESSAGE_LENGTH = Records.LOG_OVERHEAD;
 
-    private RecordValidationStats recordValidationStats;
+    private final RecordValidationStats recordValidationStats;
     private RecordsSend convertedRecordsWriter;
-    private Iterator<ConvertedRecords<?>> convertedRecordsIterator;
+    private final Iterator<ConvertedRecords<?>> convertedRecordsIterator;
```
Comment (mimaison) [inline clients/src/main/java/org/apache/kafka/common/record/LazyDownConversionRecordsSend.java:40]: We can swap this line with the one above to group all final fields

File: jmh-benchmarks/src/main/java/org/apache/kafka/jmh/acl/StandardAuthorizerUpdateBenchmark.java [hunk 5]
```diff
@@ -59,17 +59,15 @@
 @BenchmarkMode(Mode.AverageTime)
 @OutputTimeUnit(TimeUnit.MILLISECONDS)
 public class StandardAuthorizerUpdateBenchmark {
-    @Param({"25000", "50000", "75000", "100000"})
-    private int aclCount;
+    private static final Random RANDOM = new Random(System.currentTimeMillis());
     private final String resourceNamePrefix = "foo-bar35_resource-";
```
Comment (mimaison) [inline jmh-benchmarks/src/main/java/org/apache/kafka/jmh/acl/StandardAuthorizerUpdateBenchmark.java]: Can we move this field down?

**Review by mimaison — COMMENTED** [review]
Thanks, I left a few suggestions. You also need to rebase since I merged https://github.com/apache/kafka/pull/15128

File: clients/src/main/java/org/apache/kafka/common/record/LazyDownConversionRecordsSend.java (hunk 4 above)
Comment (divijvaidya) [reply clients/src/main/java/org/apache/kafka/common/record/LazyDownConversionRecordsSend.java:40]: done

File: clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java (hunk 3 above)
Comment (divijvaidya) [reply clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java:69]: done

File: jmh-benchmarks/src/main/java/org/apache/kafka/jmh/acl/StandardAuthorizerUpdateBenchmark.java (hunk 5 above)
Comment (divijvaidya) [reply jmh-benchmarks/src/main/java/org/apache/kafka/jmh/acl/StandardAuthorizerUpdateBenchmark.java]: done

**Comment by divijvaidya** [comment]
I have rebased from trunk. Apologies that I didn't create a separate commit which would have made it easier to review the diff. Is there a better way to do it?

**Review by mimaison — APPROVED** [review]
LGTM, thanks

**Comment by divijvaidya** [comment]
The compilation is successful. This PR doesn't impact any runtime and hence, tests failing in CI can be ignored.

**Pull request merged**